src/coreclr/src/jit/importer.cpp

```diff
@@ -993,6 +1000,111 @@ GenTreeCall::Use* Compiler::impPopCallArgs(unsigned count, CORINFO_SIG_INFO* sig
     return argList;
 }
 
+static bool TypeIs(var_types type1, var_types type2)
+{
+    return type1 == type2;
+}
+
+// Check if type1 matches any type from the list.
+template <typename... T>
+static bool TypeIs(var_types type1, var_types type2, T... rest)
+{
+    return TypeIs(type1, type2) || TypeIs(type1, rest...);
+}
+
```

Comment:
Seems like these are generally useful and should be in a header somewhere?

Reply:
We have `GenTree::TypeIs` and it is usually sufficient, we don't have many places where we compare types not from tree nodes.
I did not want to make this function general available because it is a little confusing that the first argument is special there, but I am open to other opinions. I thought it would be nice to make it a member function, like `var_types::TypeIs(this, types to compare this with)` but `var_types` is not a class, it is a enum.